Grievance_Backend/grievance/src/main/java/com/grievance/controller/TicketController.java

```diff
@@ -37,21 +48,63 @@ public class TicketController {
    */
   @PostMapping("/addTicket")
   public ResponseEntity<?> saveTicket(
-     @RequestBody final TicketInDto ticketInDto) {
-     Optional<TicketOutDto> optionalTicketOutDto =
-         ticketService.saveTicket(ticketInDto);
+            @RequestBody final TicketInDto ticketInDto) {
+    Optional<TicketOutDto> optionalTicketOutDto =
+            ticketService.saveTicket(ticketInDto);
     return new ResponseEntity<>(optionalTicketOutDto, HttpStatus.CREATED);
   }
 
   /**
    * controller method to return list of all tickets.
    *
-   * @return Responseentity with optional of list of all tickets.
+   * @param departmentName
+   * @param email
+   * @param password
+   * @return ResponseEntity with optional of list of all tickets.
    */
   @GetMapping("/listAllTickets")
-  public ResponseEntity<?> listAllTickets() {
-      Optional<List<TicketOutDto>>
-      optional = ticketService.listOfAllTickets();
-      return new ResponseEntity<>(optional, HttpStatus.ACCEPTED);
+  public ResponseEntity<?> listAllTickets(
+    @RequestParam(required = false) final String departmentName,
+    @RequestHeader final String email,
+    @RequestHeader final String password
+  ) {
+    try {
+      if (departmentName == null) {
```

Comment:
check null with Objects.isNull() instead of == null

Reply:
the changes are updated.